snap/snapcraft.yaml

```diff
@@ -68,11 +68,10 @@ plugs:
     interface: docker
   opengl:
   # For nvidia userspace libs support #
-  graphics-core22:
+  gpu-2404:
     interface: content
-    target: $SNAP/graphics
-    # Currently, this is the only relevant provider, but cannot be default as not supported on all archs #
-    #default-provider: nvidia-core22
+    target: $SNAP/gpu-2404
+    default-provider: mesa-2404
```

Comment:
I believe setting mesa-2404 as the default provider results in installing that snap together with the Docker snap, even if a GPU isn't present or wanted.

https://snapcraft.io/docs/auto-connection-mechanism#heading--default--provider

Reply:
Thanks @farshidtz 

The default wasn't specified previously, so I think it's fine to remove that.  I've left it in as/with a comment.

Reply:
Ahh, the problem here seems to be that gpu wrapper script errors if the interface isn't connected.

https://github.com/canonical/gpu-snap/blob/main/bin/gpu-2404-wrapper

Here is the example: https://github.com/canonical/docker-snap/actions/runs/12677452152/job/35333257896#step:7:23

Hmmm.  Will have a think about how best to handle that..